### DISCUSSION TITLE
Logout request to Keycloak getting cancelled during redirection

### DISCUSSION THREAD

**Discussion by vishsum** [keycloak/keycloak#24203] [Q&A]
![image](https://github.com/keycloak/keycloak/assets/140374126/144e13d1-ddca-481c-b630-a59410b98d5c)

**Issue Snapshot Description:**

We have been using Keycloak (version 21.0.2) for authentication and authorization in our React application, and for the most part, it has been working smoothly. However, we've encountered an issue where the logout request to Keycloak and the redirection with keycloak is getting cancelled, albeit not frequently.

**Issue Details:**

**Frequency:** This issue is sporadic and does not occur with every logout attempt. However, when it does happen, it disrupts the user experience.

**Logout Flow:** We have implemented a standard logout flow in our React application, where the user clicks on the logout button, and the application sends a request to Keycloak to terminate the session.

**Issues with Multiple Redirections:** In some scenarios, When we are using Keycloak with Okta integration. Our authentication flow includes default redirections between our React application, Keycloak, and Okta. Here also the logout process is getting cancelled during these intermediary redirections.

**Symptoms:** When this issue occurs, it appears as though the logout request is initiated, but before it is completed, it gets cancelled. Users are not logged out of the application, and they remain authenticated, which poses a security concern.

**Seeking Community Input:**

Given the sporadic nature of this issue, we are reaching out to the GitHub community for any insights, suggestions, or experiences related to similar problems with Keycloak version 21.0.2. Has anyone else encountered a situation where logout requests to Keycloak are getting cancelled intermittently? If so, how did you address this issue, and what steps did you take to resolve it?

Please share your thoughts, experiences, or potential solutions. Any help would be greatly appreciated in resolving this logout cancellation issue.

**Comment by moriyasror** [comment]
@vishsum  we also experience that, did you found solution

**Reply by vishsum** [reply]
We resolved the issue by upgrading Keycloak version from 21.0.2 to 22.0.5. Ensure that your session timeout matches with your IdP (Okta).

**Comment by rensrud** [comment]
Have same issue in Keycloak 24.0.1. It also happens on 25.0.6 but less often.

**Comment by danielFesenmeyer** [comment]
Seems like this issue can be reproduced with the Account UI log out. We observed a similar behavior in our (Selenium/Playwright) integration tests: Sometimes the user was still logged in after clicking "Log out". We stabilized the tests by just removing all cookies instead of using the button.
We could not reproduce it manually, so we assumed it's a timing issue with "our tests clicking too fast".

But it seems the mentioned problem is related to our observed behavior.

**Reply by danielFesenmeyer** [reply]
Btw, we're still on Keycloak 24 currently. We also observed that with previous versions.

**Reply by codespearhead** [reply]
Can you confirm if this issue is still reproducible in the latest stable release of Keycloak, which currently is version 26.0.1? Only the latest release will receive updates [1][2].

[3] might have fixed it.

***

[1] https://github.com/keycloak/keycloak/discussions/29126
[2] https://github.com/keycloak/keycloak/discussions/29126#discussioncomment-9257666
[3] https://github.com/keycloak/keycloak/pull/32130

**Reply by danielFesenmeyer** [reply]
I don't know, I'd have to build a reproducer first. I'll let you know when I have found the time to do that.
